Packs/Logzio/Integrations/Logzio/Logzio.py

```diff
@@ -211,7 +222,11 @@ def fetch_incidents(client, last_run, search, severities, first_fetch_time):
         next_run["last_fetch"] = max(start_query_time, time.time() - ONE_HOUR)
     raw_events = client.fetch_triggered_rules(search=search, severities=severities,
                                               start_time=start_query_time)
-    for event in raw_events.get("results", []):
+
+    # Filter events if any of the filter tags are in an event
+    tag_filtered_events = [x for x in raw_events.get("results", []) if found_tag_in_event(x, filter_tags)]
```

Comment:
Can you move this logic to be inside the main loop ( line 229 )
At the current state you are running twice on all the events
You can move it to be in the main loop and if it return false then continue (move to the next event)